src/main/java/net/eldeen/dropwizard/CloudFormationFacade.java

```diff
@@ -0,0 +1,80 @@
+import static com.amazonaws.services.cloudformation.model.ResourceStatus.UPDATE_IN_PROGRESS;
+
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.cloudformation.AmazonCloudFormation;
+import com.amazonaws.services.cloudformation.AmazonCloudFormationClient;
+import com.amazonaws.services.cloudformation.model.DescribeStackResourceRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStackResourceResult;
+import com.amazonaws.services.cloudformation.model.ResourceSignalStatus;
+import com.amazonaws.services.cloudformation.model.ResourceStatus;
+import com.amazonaws.services.cloudformation.model.SignalResourceRequest;
+import com.amazonaws.util.EC2MetadataUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CloudFormationFacade {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CloudFormationFacade.class);
+
+  private final String stackName;
+  private final Region awsRegion;
+
+  CloudFormationFacade(String stackName, String awsRegion) {
+    this.stackName = stackName;
+    this.awsRegion = Region.getRegion(Regions.valueOf(awsRegion.toUpperCase().replace('-', '_')));
+  }
+
+  boolean isInAws() {
+    String instanceId = getInstanceId();
+    return instanceId != null && instanceId.trim().length() > 0;
+  }
+
+  boolean isUpdateInProgress(String resourceName) {
+    AmazonCloudFormation client = new AmazonCloudFormationClient();
+    try {
+      client.setRegion(awsRegion);
+
+      DescribeStackResourceResult asgResource =
+          client.describeStackResource(new DescribeStackResourceRequest()
+                                           .withStackName(stackName)
+                                           .withLogicalResourceId(resourceName));
+      ResourceStatus resourceStatus =
+          ResourceStatus.fromValue(asgResource.getStackResourceDetail().getResourceStatus());
+
+      return resourceStatus == CREATE_IN_PROGRESS
+          || resourceStatus == UPDATE_IN_PROGRESS;
+    }
+    catch (Exception e) {
+      LOGGER.error("There was a problem describing " + resourceName + " in stack " + stackName, e);
+      return false;
+    }
+    finally {
+      client.shutdown();
+    }
+  }
+
+  void signalResource(String resourceName) {
+    AmazonCloudFormation client = new AmazonCloudFormationClient();
+    try {
+      client.setRegion(awsRegion);
+      client.signalResource(new SignalResourceRequest()
+                                .withUniqueId(getInstanceId())
+                                .withLogicalResourceId(resourceName)
+                                .withStackName(stackName)
+                                .withStatus(ResourceSignalStatus.SUCCESS));
+    }
+    catch (Exception e) {
+      LOGGER.error("There was a problem signaling " + resourceName + " in stack " + stackName, e);
+    }
+    finally {
+      client.shutdown();
+    }
+  }
+
+  private String getInstanceId() {
```

Comment:
This is probably a bad idea putting this in here since `EC2Metadata` has nothing to do w/ cloud formation.